lib/HttpClient.js

```diff
@@ -89,14 +89,68 @@ function defaultUserAgent() {
 }
 
 
+/**
+ * A function that handles the issuing of the raw request through the
+ * underlying http core modules. This function is used as the target of
+ * retry/backoff mechanism for retrying on connection timeouts.
+ *
+ * The callback provided to this function is invoked with the following
+ * signature:
+ *      function(err, req) { ... }
+ * 1) `err` can only ever be a ConnectionTimeout. If this happens, req is
```

Comment:
It comes from here:
https://github.com/restify/clients/blob/415bac982f007ca3cd0de9b24df00d84651f4edf/lib/HttpClient.js#L266

If you meant you don't understand how it can _only_ be ConnectionTimeout error, I actually agree, that's not true. I'll update the text to note that the first callback would be triggered with any errors that could happen before establishing a connection.

Reply:
Yes, sorry, somehow the word "only" didn't make it in my comment, but that's what I meant :)